modin/engines/base/frame/data.py

```diff
@@ -1672,25 +1672,7 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         if isinstance(other, type(self)):
             other = [other]
 
-        is_aligning_applied = False
-        for i in range(len(other)):
-            if (
-                len(self._partitions) != len(other[i]._partitions)
-                and len(self.axes[0]) == len(other[i].axes[0])
-                and axis == 0
-            ):
-                is_aligning_applied = True
-                self._partitions = self._frame_mgr_cls.map_axis_partitions(
-                    axis, self._partitions, lambda df: df
-                )
-                other[i]._partitions = other[i]._frame_mgr_cls.map_axis_partitions(
-                    axis, other[i]._partitions, lambda df: df
-                )
```

Comment:
As I understood we will make repartitioning when `joined_index` isn't equal to old left/right indexes. This block of code was a special case of this? Am I right?

Reply:
Not completely, most likely we simply do not have tests for this case.

Reply:
We have test for this case (you can find this [here](https://github.com/modin-project/modin/blob/master/modin/pandas/test/dataframe/test_binary.py#L195))

Reply:
For binary operations - no problem.

I also found the case, that is not tested:

```python
    data = [0, 1, 2, 3, 4, 5]
    modin_df1, pandas_df1 = create_test_dfs({"a": data, "b": data})
    modin_df, pandas_df = modin_df1.loc[:2], pandas_df1.loc[:2]

    modin_df2 = modin_df.append(modin_df)
    pandas_df2 = pandas_df.append(pandas_df)
    
    modin_df2["c"] = modin_df1["b"]
    print(modin_df2)
Exception: Internal Error. Please email bug_reports@modin.org with the traceback and command that caused this error.
Row lengths: 6 != 9
```

Reply:
@anmyachev Ping me please when you will be ready for review.